approximate_string_matching/four_russians_helpers.py

```diff
@@ -0,0 +1,260 @@
+import math
+
+lcs_delete_cost_function = lambda a: 1
+lcs_insert_cost_function = lambda b: 1
+lcs_substitute_cost_function = lambda a, b: 0 if (a == b) else 2
+
+class FourRussiansHelpers:
+  def __init__(
+      self,
+      delete_cost_function,
+      insert_cost_function,
+      substitute_cost_function):
+    self.delete_cost_function = delete_cost_function
+    self.insert_cost_function = insert_cost_function
+    self.substitute_cost_function = substitute_cost_function
+
+  def prepare_parameters(self, text_1, text_2):
+    A = sorted(list(set(list(text_1[1:] + text_2[1:]))))
+
+    def get_parameter(text_1):
+      if len(text_1) == 1:
+        return 1
+      return int(math.log2(len(text_1) - 1))
+
+    def get_step_size_bound():
+      if len(A) == 0:
+        return (0, 0)
+      I = max([self.insert_cost_function(letter_idx) for letter_idx in A])
+      D = max([self.delete_cost_function(letter_idx) for letter_idx in A])
+      return max(I, D)
+
+    m = get_parameter(text_1)
+    step_size_bound = get_step_size_bound()
+
+    text_1_mod = (len(text_1) - 1) % m
+    if text_1_mod != 0:
+      text_1 += '#' * (m - (text_1_mod))
+
+    text_2_mod = (len(text_2) - 1) % m
+    if text_2_mod != 0:
+      text_2 += '#' * (m - (text_2_mod))
+
+    return m, A, step_size_bound, text_1, text_2
+
+  @staticmethod
+  def get_all_strings(m, A, dumb_letter):
+    """ Returns all possible strings of a given length """
+
+    sub_length_results = [[] for _ in range(m + 1)]
+    sub_length_results[0] = []
+    sub_length_results[1] = A
+
+    for size in range(2, m + 1):
+      sub_length_results[size] = []
+      for left_substring in sub_length_results[1]:
+        for right_substring in sub_length_results[size - 1]:
+          sub_length_results[size].append(left_substring + right_substring)
```

Comment:
More Pythonic would be something like
sub_length_results[size] = [left_substring + right_substring for left_substring in sub_length_results[1] for right_substring in sub_length_results[size - 1]]